From 7d9acd12ae5d1773312dc5b532fa5c0d30486ed3 Mon Sep 17 00:00:00 2001
From: Dario Binacchi <dario.binacchi@amarulasolutions.com>
Date: Mon, 9 Mar 2026 08:17:36 +0100
Subject: [PATCH 1/4] feat(st): add extended bootloader GUID

The Extended Boot Loader Partition (XBOOTLDR) is a standard defined by
the Discoverable Partitions Specification (DPS) to host boot loader
resources outside of the EFI System Partition [1], [2].

Defining this GUID (bc13c2ff-59e6-4262-a352-b275fd6f7172) allows TF-A
to correctly identify these partitions.

[1] https://uapi-group.org/specifications/specs/discoverable_partitions_specification/#extended-boot-loader-partition:~:text=UEFI%20Specification.-,Extended%20Boot%20Loader%20Part>
[2] https://uapi-group.org/specifications/specs/boot_loader_specification/

Change-Id: I291cd0cf76442d52b2ed6aa6f44943ce21f942fd
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
Upstream-Status: Backport [https://git.trustedfirmware.org/plugins/gitiles/TF-A/trusted-firmware-a/+/e9a8085667b849bcba562aaf457089d3f63888d7]
---
 plat/st/common/include/stm32mp_efi.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/st/common/include/stm32mp_efi.h b/plat/st/common/include/stm32mp_efi.h
index af9165f1daa5..63e1bd86da03 100644
--- a/plat/st/common/include/stm32mp_efi.h
+++ b/plat/st/common/include/stm32mp_efi.h
@@ -13,4 +13,8 @@
 	EFI_GUID(0x19d5df83U, 0x11b0U, 0x457bU, \
 		 0xbeU, 0x2cU, 0x75U, 0x59U, 0xc1U, 0x31U, 0x42U, 0xa5U)
 
+#define XBOOTLDR_GUID \
+	EFI_GUID(0xBC13C2FFU, 0x59E6U, 0x4262U, \
+		 0xA3U, 0x52U, 0xB2U, 0x75U, 0xFDU, 0x6FU, 0x71U, 0x72U)
+
 #endif /* STM32MP_EFI_H */
-- 
2.43.0

